indexing-service/src/main/java/org/apache/druid/indexing/worker/shuffle/ShuffleMetrics.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.worker.shuffle;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class ShuffleMetrics
+{
+  private final AtomicReference<ConcurrentHashMap<String, PerDatasourceShuffleMetrics>> datasourceMetrics =
+      new AtomicReference<>();
+
+  public ShuffleMetrics()
+  {
+    datasourceMetrics.set(new ConcurrentHashMap<>());
+  }
+
+  public void shuffleRequested(String supervisorTaskId, long fileLength)
+  {
+    datasourceMetrics
+        .get()
+        .computeIfAbsent(supervisorTaskId, k -> new PerDatasourceShuffleMetrics()).accumulate(fileLength);
+  }
+
+  public Map<String, PerDatasourceShuffleMetrics> snapshot()
```

Comment:
should this be renamed to `snapshotAndReset` or may be just `reset` ?
```
public Map<String, PerDatasourceShuffleMetrics> reset()
{
   return Collections.unmodifiableMap(datasourceMetrics.getAndSet(new ConcurrentHashMap<>()));
}
```

Reply:
Sounds good. Changed to `snapshotAndReset()` since it sounds more intuitive to me.